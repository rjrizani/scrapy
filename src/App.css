.App {
  background-color: #282c34;
  text-align: left;
}

a {
  color: #FFF;
}
a:hover {
   color: #00F
}

* {
  box-sizing: border-box;
}

body {
  margin: 0;
  }

/* Style the header */
.header {
  background-color: #61dafb;
  padding-left: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
/* Style the footer */
.footer {
  background-color: #f1f1f1;
  padding: 10px;
  padding-left: 20px;
  text-align: center;
}

/* Responsive layout - makes the three columns stack on top of each other instead of next to each other */
@media screen and (max-width:600px) {
  .column {
    width: 100%;
    padding: 25px;
    
  }
}
.App-header {
  background-color: #282c34;
  min-height: 20vh;
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-body {
  background-color: #61dafb;
  padding-left: 40px;
  min-height: 50vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}



.App-link {
  color: #61dafb;
}


