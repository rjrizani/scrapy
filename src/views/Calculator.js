import React from 'react';


const scaleNames = {
  j: 'Jumlah Mata Pelajaran',
  t: 'Total Biaya (Rupiah)'
};
//fungsi untuk logika matematika yang diterapkan
function toJumlah(Total) {
  return Total / 150000;
}

function toTotal(Jumlah) {
  return Jumlah * 150000;
}

function tryConvert(harga, convert) {
  const input = parseFloat(harga);
  if (Number.isNaN(input)) {
    return '';
  }
  const output = convert(input);
  const rounded = Math.round(output * 1000) / 1000;
  return rounded.toString();
}


class HargaInput extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    this.props.onHargaChange(e.target.value);
  }

  render() {
    const harga = this.props.harga;
    const scale = this.props.scale;
    return (
      <fieldset>
        <legend>{scaleNames[scale]}:</legend>
        <input value={harga}
               onChange={this.handleChange} />
      </fieldset>
    );
  }
}

//Lifting state up konsep di terapkan pada component Calculator
class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.handleJumlahChange = this.handleJumlahChange.bind(this);
    this.handleTotalChange = this.handleTotalChange.bind(this);
    this.state = {harga: '', scale: 'j'};
  }

  handleJumlahChange(harga) {
    this.setState({scale: 'j', harga});
  }

  handleTotalChange(harga) {
    this.setState({scale: 't', harga});
  }

  render() {
    const scale = this.state.scale;
    const harga = this.state.harga;
    const Jumlah = scale === 't' ? tryConvert(harga, toJumlah) : harga;
    const Total = scale === 'j' ? tryConvert(harga, toTotal) : harga;

    return (
      <div className="App-body">
            <header >
            Kalkulator Biaya Kursus
            </header> <br/>
        <HargaInput
          scale="j"
          harga={Jumlah}
          onHargaChange={this.handleJumlahChange} />
        <HargaInput
          scale="t"
          harga={Total}
          onHargaChange={this.handleTotalChange} />
        <p> Biaya kursus dihitung dengan biaya Rp.150.000,00 per 3 kali pertemuan untuk 1 mata pelajaran</p>
      </div>
    );
  }
}

export default Calculator;
