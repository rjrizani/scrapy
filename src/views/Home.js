import React from 'react';

class Home extends React.Component {
  
  render() {
     return (
        <div>
           <Header/>
           <Content/>
        </div>
     );
  }
}
class Header extends React.Component {
  render() {
     return (
        <div className="header">
        <h1> Home</h1> 
        
        </div>
     );
  }
}
class Content extends React.Component {
  render() {
    const judul = ("Masterscrapy adalah layanan kursus online yang berbasis pada pendekatan konsultasi. Mengapa belajar dimasterscarpy:");
    const mamfaat = ["Bisa belajar dimana saja dan kapan saja.", 
                      "Sistem belajar online bukan cuma mendengarkan saja", 
                      "Konsep konsultasi yang membuat siswa berpikir kritis", 
                      "Pengajar profesional dari S1-S2.", 
                      "Kursus bisa online dan offline."];
    
    const listItems = mamfaat.map((faat) => <li key={faat}>{faat}</li>);

     return (
        <div className="App-body">
        <h2>{judul}</h2> 
      
        <ul>{listItems}</ul>
            <h2>Program Rahasia Pintar</h2>
            <button>SMA</button> <br/>
            <button>SMP</button> <br/>
            <button>SD</button>  <br/>
        </div>
     );
  }
}

export default Home;
