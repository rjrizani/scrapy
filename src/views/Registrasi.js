import React from 'react';

//Controlled Components form diterapkan disini ditandai dengan pengaturan state
//constructor
//fungsi bind(this) agar referensi nilai tidak berubah saat dieksekusi.
//fungsi handleSubmit() berguna untuk menerima submit form dan meneksekusi logika didalamnya
//berupa alert data form
//fungsi handleInputChange() untuk mengubah state awal yang diterima dan diinput oleh user.
//preventDefault berfungsi mencegah terrefresnya brosur ketika data disubmit
export default class Registrasi extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            Nama: '',
            NomorWa: '',
            Umur: 4,
            Alamat: 'tulis alamatnya yah',
            Sosial: '',
            Wali: '',
            NoWali: ''
        };

      this.handleInputChange = this.handleInputChange.bind(this);
      this.handleSubmit = this.handleSubmit.bind(this);
    }


  
     handleInputChange(event) {
        const target = event.target;
        const value = target.name === 'xx' ? target.checked : target.value;
        const name = target.name;
        
        this.setState({
           [name]: value});
     }
  
     handleSubmit(event) {
        alert('Nama: ' + this.state.Nama +
        '   Nomor Wa: ' + this.state.NomorWa +
        '   Umur: ' + this.state.Umur +
        '   Alamat: ' + this.state.Alamat +
        '  Pendidikan: ' + this.state.Pendidikan +
        '  IG/FB: ' + this.state.Sosial +
        '  Nama Wali: ' + this.state.Wali +
        '  Nomor Wali: ' + this.state.NoWali 
        );
        
        event.preventDefault();
        }

    render() {
        return (
            <div className="App-body" >

            <header > <br/>
                <h1>Registrasi Form</h1>
            </header> 

             <form className="App-body" onSubmit={this.handleSubmit}>
                Nama: 
                <input type="text" name="Nama" value={this.state.Nama} onChange={this.handleInputChange} /> 
                Nomor HP/WA: 
                <input type="text" name="NomorWa" value={this.state.NomorWa} onChange={this.handleInputChange} /> 
                Umur(Tahun): 
                <input type="number" name="Umur" value={this.state.Umur} onChange={this.handleInputChange} /> 
                Alamat: 
                <textarea name="Alamat" value={this.state.Alamat} onChange={this.handleInputChange} /> 
                Pendidikan: 
                <select name="Pendidikan" value={this.state.Pendidikan} onChange={this.handleInputChange}> 
                    <option pendidikan="SMA">SMA</option>
                    <option pendidikan="SMP">SMP</option>
                    <option pendidikan="SD">SD</option>
                </select>
                Akun media sosial (Instagram/Facebook): 
                <input type="type" name="Sosial" value={this.state.Sosial} onChange={this.handleInputChange} /> 
                Nama Orang Tua/Wali: 
                <input type="text" name="Wali" value={this.state.Wali} onChange={this.handleInputChange} /> 
                Nomor Orang Tua/Wali: 
                <input type="text" name="NoWali" value={this.state.NoWali} onChange={this.handleInputChange}/> <br/>
                <input type="submit" value="Submit" />
             </form><br/>  
            </div>
        )
    }
    
}
