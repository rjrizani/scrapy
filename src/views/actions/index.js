// fungsi memaanggil state case pada todo
const _add = () => ({
    type: 'ADD'
})

const _substract = () => ({
    type: 'SUBSTRACT'
})

//berfungsi memberikan delay 1 detik ketika couter ditambah atau dikurangin
export function incrementAsync() {
    return async (dispatch) => {
        //jika ingin mengunakan api
        // const response = await fetch('http://newsapi.org.......');
        // console.log(response);
        
        // dispatch({
        //     type: 'GET_DATA',
        //     payload: response.json()
        // })
        
        
        setTimeout(() => {
            dispatch(_add());
        }, 1000)
    }
}
;
export function decrementAsync() {
       return async (dispatch) => {
             
        setTimeout(() => {
            dispatch(_substract());
        }, 1000)
    }
}