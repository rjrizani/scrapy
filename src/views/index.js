export {default as Home} from './Home';
export {default as Registrasi} from './Registrasi';
export {default as Harapan} from './Harapan';
export {default as Calculator} from './Calculator';
