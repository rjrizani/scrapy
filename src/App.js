import React from 'react';
import './App.css';

import { 
  BrowserRouter as Router,
  Switch,
  Route,
  Link
 } from 'react-router-dom';

import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import reducer from './reducers';
import {Home, Registrasi, Harapan, Calculator} from './views';
//state management menggunakan redux-thunk 
//state dibuat global yang disimpan dalam varibel store 
//kemudian dikembalikan ke component Harapan
const store = createStore(reducer, applyMiddleware(thunk))

//component App berfunsi sebagai parent untuk menjalakan compponent child 
//(Home, Registrasi, Harapan,Calculator) dengan menerapkan fungsi router
// Link to="/" dan exact path berfungsi membuat arah tampilan awal ketika web dipanggil
//
 function App() {
      
   return(
     //fungsi router diterapkan disini
     <Router>
      <div className="App">
        <header className="App-header">
          <ul>
            <li><Link to="/"> Home </Link></li>
            <li><Link to="/registrasi"> Registrasi </Link></li>
            <li><Link to="/harapan"> Harapan </Link></li>
            <li><Link to="/calculator"> Harga </Link></li>
          </ul>
        </header>
        <Switch>
          <Route exact path="/">   
            <Home/>
          </Route>
          <Route path="/registrasi">
            <Registrasi/>
          </Route> 
          <Route path="/harapan">
            <Provider store={store}>
              <Harapan/>
            </Provider>
          </Route>
          <Route path="/calculator">
            <Calculator/>
          </Route>
        </Switch>

      </div>  

      <div className="footer">
          <p>ig.masterscrapy</p>
          <p>speaksma@gmail.com</p>
       </div>
     </Router>
    

   );
 }

 export default App;
