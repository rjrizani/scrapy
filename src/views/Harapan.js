import React,{useEffect} from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { incrementAsync, decrementAsync } from './actions';


export default function Harapan() {
    //Lifecycle methond dengan menggunakan useEffect function yang berguna seperti componentDidMount 
    //untuk auto turun kebawah page.
    // useEffect(()=>{
    // window.scrollTo(0,window.innerHeight,'smooth');
    // },[]);

    //fungsi useEffect digunakan seperti konsep lifecylce ketika masuk ke page Harapan title berubah
    //ketika keluar dari page harapan title kembali ke awal.
    //side effect hook
    useEffect(()=>{
      document.title = `Motivasi Positif: ${counter}`;
      return () =>{
        document.title = `Rahasia Pintar`; }
    });
 
    const dispatch = useDispatch();
    const counter = useSelector(state => state.counter);

    const _add = () => {
        dispatch(incrementAsync());
    }

    const _substract = () => {
        dispatch(decrementAsync());
    }

  return (
    <div className="App-body">
        <header > <br/>
        Pemberi Harapan 
        </header> <br/>
        <textarea style={{width: '300px',height: '100px'}} type="text" placeholder="Ketik harapanmu disini" /> <br/>
      <div>Jumlah Pemberi Harapan: {counter}</div>
        <div style={{ display: 'flex', flexDirection: 'row' }}>
        <button style={{ width: 100 }} onClick={_substract}>Substract</button>
        <button style={{ width: 100 }} onClick={_add}>Add</button>
      </div>
      <h3>Bahaya Putus Asa dan Manfaat Menjaga Harapan</h3> 
      <p style={{paddingRight: '80px'}}>
      HARAPAN adalah sinar terang yang membelah gelapnya cobaan. 
      Harapan adalah pembimbing yang cerdas dalam pekatnya kesulitan, 
      ilmu yang menjadi petunjuk dalam sulitnya permasalahan, 
      penguasa perkasa yang mendorong tekad saat terjadi penurunan dan
       pendongkrak semangat saat terjadi stagnasi. <br/>

Sebaliknya, putus asa adalah penyakit mematikan bagi jiwa manusia. 
Putus asa adalah penyakit yang paling gigih dilawan oleh syariat islam, 
karena hidup adalah gerak dan dinamika. Antusias adalah kudanya. Ia adalah kendaraan 
cita-cita untuk meraih keluhuran dalam berbagai bidang kehidupan. Sementara putus asa 
adalah musuh bebuyutan yang akan melemahkan tekad, menghentikan langkah dan meniadakan harapan.
(https://www.islampos.com/bahaya-putus-asa-dan-manfaat-menjaga-harapan-35339/) 
<br/>
Sehingga penting bagi kita untuk memiliki harapan untuk memotivasi diri untuk terus berkarya dan berkembang
kearah yang positif.
      </p>
    </div>
  );
}



