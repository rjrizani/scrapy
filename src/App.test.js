import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import App from './App';
import { Registrasi, Harapan, Calculator } from './views';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

//snapshot testing 
test('fungsi handel change',()=>{
  const component = renderer.create(
    <Calculator/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.getInstance().handleJumlahChange();
  tree=component.toJSON();
  expect(tree).toMatchSnapshot();
  
  component.getInstance().handleTotalChange();
  tree=component.toJSON();
  expect(tree).toMatchSnapshot();
  
});

test('x',()=>{
  const div = document.createElement("div");
  ReactDOM.render(<button></button>, div);
})

// test('fungsi handel input registrasi',()=>{
//     const component = renderer.create(
//       <Registrasi/>
//     );
  
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
  
//     component.getInstance().handleSubmit();
//     tree=component.toJSON();
//     expect(tree).toMatchSnapshot();
   
//   });

  //redux #Middleware test
  const thunk = ({ dispatch, getState }) => next => action => {
    if (typeof action === 'function') {
      return action(dispatch, getState)
    }
  
    return next(action)
  }

  const create = () => {
    const store = {
      getState: jest.fn(() => ({})),
      dispatch: jest.fn()
    }
    const next = jest.fn()
  
    const invoke = action => thunk(store)(next)(action)
  
    return { store, next, invoke }
  }

  test('passes through non-function action', () => {
    const { next, invoke } = create()
    const action = { type: 'TEST' }
    invoke(action)
    expect(next).toHaveBeenCalledWith(action)
  })
  
  test('calls the function', () => {
    const { invoke } = create()
    const fn = jest.fn()
    invoke(fn)
    expect(fn).toHaveBeenCalled()
  })
  
  test('passes dispatch and getState', () => {
    const { store, invoke } = create()
    invoke((dispatch, getState) => {
      dispatch('TEST DISPATCH')
      getState()
    })
    expect(store.dispatch).toHaveBeenCalledWith('TEST DISPATCH')
    expect(store.getState).toHaveBeenCalled()
  })

  