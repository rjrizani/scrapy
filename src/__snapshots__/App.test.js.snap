// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fungsi handel change 1`] = `
<div
  className="App-body"
>
  <header>
    Kalkulator Biaya Kursus
  </header>
   
  <br />
  <fieldset>
    <legend>
      Jumlah Mata Pelajaran
      :
    </legend>
    <input
      onChange={[Function]}
      value=""
    />
  </fieldset>
  <fieldset>
    <legend>
      Total Biaya (Rupiah)
      :
    </legend>
    <input
      onChange={[Function]}
      value=""
    />
  </fieldset>
  <p>
     Biaya kursus dihitung dengan biaya Rp.150.000,00 per 3 kali pertemuan untuk 1 mata pelajaran
  </p>
</div>
`;

exports[`fungsi handel change 2`] = `
<div
  className="App-body"
>
  <header>
    Kalkulator Biaya Kursus
  </header>
   
  <br />
  <fieldset>
    <legend>
      Jumlah Mata Pelajaran
      :
    </legend>
    <input
      onChange={[Function]}
    />
  </fieldset>
  <fieldset>
    <legend>
      Total Biaya (Rupiah)
      :
    </legend>
    <input
      onChange={[Function]}
      value=""
    />
  </fieldset>
  <p>
     Biaya kursus dihitung dengan biaya Rp.150.000,00 per 3 kali pertemuan untuk 1 mata pelajaran
  </p>
</div>
`;

exports[`fungsi handel change 3`] = `
<div
  className="App-body"
>
  <header>
    Kalkulator Biaya Kursus
  </header>
   
  <br />
  <fieldset>
    <legend>
      Jumlah Mata Pelajaran
      :
    </legend>
    <input
      onChange={[Function]}
      value=""
    />
  </fieldset>
  <fieldset>
    <legend>
      Total Biaya (Rupiah)
      :
    </legend>
    <input
      onChange={[Function]}
    />
  </fieldset>
  <p>
     Biaya kursus dihitung dengan biaya Rp.150.000,00 per 3 kali pertemuan untuk 1 mata pelajaran
  </p>
</div>
`;
